### PR TITLE
[type:feat]Selector module adapts to namespace transformation

### PR DESCRIPTION
**Pull request by xcsnx**
Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java [hunk 1]
```diff
@@ -71,35 +80,39 @@ public interface SelectorMapper extends ExistProvider {
...
     /**
      * Find by plugin id list.
      *
-     * @param pluginIds the plugin ids
+     * @param pluginIds   the plugin ids
+     * @param namespaceId namespaceId.
      * @return the list
      */
-    List<SelectorDO> findByPluginIds(List<String> pluginIds);
+    List<SelectorDO> findByPluginIds(@Param("list") List<String> pluginIds, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java:96]: add new method

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java [hunk 2]
```diff
@@ -71,35 +80,39 @@ public interface SelectorMapper extends ExistProvider {
...
     /**
      * select select by name.
      *
-     * @param name the name
+     * @param name        the name
+     * @param namespaceId namespaceId.
      * @return selector do list
      */
-    List<SelectorDO> selectByName(String name);
+    List<SelectorDO> selectByName(String name, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java:105]: same questions

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/model/dto/SelectorDTO.java [hunk 3]
```diff
@@ -386,36 +414,30 @@ public void setSelectorRules(final List<RuleDTO> selectorRules) {
...
-                && Objects.equals(continued, that.continued)
-                && Objects.equals(handle, that.handle)
-                && Objects.equals(selectorConditions, that.selectorConditions)
-                && Objects.equals(matchRestful, that.matchRestful);
+        return Objects.equals(id, that.id) && Objects.equals(pluginId, that.pluginId) && Objects.equals(name, that.name)
+                && Objects.equals(matchMode, that.matchMode) && Objects.equals(type, that.type) && Objects.equals(sort, that.sort)
+                && Objects.equals(enabled, that.enabled) && Objects.equals(loged, that.loged) && Objects.equals(continued, that.continued)
+                && Objects.equals(handle, that.handle) && Objects.equals(selectorConditions, that.selectorConditions)
+                && Objects.equals(matchRestful, that.matchRestful) && Objects.equals(selectorRules, that.selectorRules)
+                && Objects.equals(namespaceId, that.namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/model/dto/SelectorDTO.java:432]: not change the style

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/model/dto/SelectorDTO.java [hunk 4]
```diff
@@ -386,36 +414,30 @@ public void setSelectorRules(final List<RuleDTO> selectorRules) {
...
     }
-    
+
     @Override
     public int hashCode() {
-        return Objects.hash(id, pluginId, name, matchMode, type, sort, enabled, loged, continued, handle,
-                selectorConditions, matchRestful);
+        return Objects.hash(id, pluginId, name, matchMode, type, sort, enabled, loged, continued, handle, selectorConditions,
+                matchRestful, selectorRules, namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/model/dto/SelectorDTO.java:438]: not change the style

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/model/entity/PluginDO.java [hunk 5]
```diff
@@ -222,6 +223,37 @@ public static PluginDO buildPluginDO(final PluginDTO pluginDTO) {
         }).orElse(null);
     }
 
+    /**
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/model/entity/PluginDO.java:226]: shoud add buildPluginDO or buildNamespacePluginDO?

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/model/entity/SelectorDO.java [hunk 6]
```diff
@@ -314,21 +340,15 @@ public boolean equals(final Object o) {
...
-                && Objects.equals(matchMode, that.matchMode)
-                && Objects.equals(type, that.type)
-                && Objects.equals(sort, that.sort)
-                && Objects.equals(enabled, that.enabled)
-                && Objects.equals(loged, that.loged)
-                && Objects.equals(continued, that.continued)
-                && Objects.equals(handle, that.handle)
-                && Objects.equals(matchRestful, that.matchRestful);
+        return Objects.equals(pluginId, that.pluginId) && Objects.equals(name, that.name) && Objects.equals(matchMode, that.matchMode)
+                && Objects.equals(type, that.type) && Objects.equals(sort, that.sort) && Objects.equals(enabled, that.enabled)
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/model/entity/SelectorDO.java:344]: not change style

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java [hunk 7]
```diff
@@ -107,55 +107,61 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
...
      *
-     * @param name       the name
-     * @param pluginName the plugin name
+     * @param name        the name
+     * @param pluginName  the plugin name
+     * @param namespaceId the namespaceId
      * @return the selector do
      */
-    SelectorDO findByNameAndPluginName(String name, String pluginName);
+    SelectorDO findByNameAndPluginName(String name, String pluginName, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:154]: same questions for current service.

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java [hunk 8]
```diff
@@ -186,7 +188,7 @@ private void removeRegister(final ApiDO apiDO) {
             }
         });
         if (CollectionUtils.isNotEmpty(selectorIds)) {
-            selectorService.delete(selectorIds);
+            selectorService.delete(selectorIds, SYS_DEFAULT_NAMESPACE_NAMESPACE_ID);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java:191]: need judge logic?

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/SelectorServiceImpl.java [hunk 9]
```diff
@@ -173,10 +175,12 @@ public String registerDefault(final SelectorDTO selectorDTO) {
     @Override
     public String registerDefault(final MetaDataRegisterDTO dto, final String pluginName, final String selectorHandler) {
         String contextPath = ContextPathUtils.buildContextPath(dto.getContextPath(), dto.getAppName());
-        SelectorDO selectorDO = findByNameAndPluginName(contextPath, pluginName);
+        // todo:[To be refactored with namespace]  Temporarily  hardcode
+        SelectorDO selectorDO = findByNameAndPluginName(contextPath, pluginName, SYS_DEFAULT_NAMESPACE_NAMESPACE_ID);
         if (Objects.isNull(selectorDO)) {
             SelectorDTO selectorDTO = SelectorUtil.buildSelectorDTO(contextPath, pluginMapper.selectByName(pluginName).getId());
             selectorDTO.setHandle(selectorHandler);
+            selectorDTO.setNamespaceId(SYS_DEFAULT_NAMESPACE_NAMESPACE_ID);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/SelectorServiceImpl.java:183]: shenyu client must add shenyu namespace id, you should get the namespace from shenyu-client, if it's null, you should use the default.

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/model/entity/PluginDO.java (hunk 5 above)
Comment (xcsnx) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/model/entity/PluginDO.java:226]: buildPluginDO

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/SelectorController.java [hunk 10]
```diff
@@ -111,10 +117,9 @@ public ShenyuAdminResult createSelector(@Valid @RequestBody final SelectorDTO se
      * @param selectorDTO selector.
      * @return {@linkplain ShenyuAdminResult}
      */
-    @PutMapping("/{id}")
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/SelectorController.java:114]: not delete `/`

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/SelectorController.java (hunk 10 above)
Comment (xcsnx) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/SelectorController.java:114]: ok

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/SelectorController.java [hunk 11]
```diff
@@ -138,12 +143,12 @@ public ShenyuAdminResult batchEnabled(@Valid @RequestBody final BatchCommonDTO b
     /**
      * delete Selectors.
      *
-     * @param ids primary key.
+     * @param batchNamespaceCommonDTO batchNamespaceCommonDTO.
      * @return {@linkplain ShenyuAdminResult}
      */
     @DeleteMapping("/batch")
-    public ShenyuAdminResult deleteSelector(@RequestBody @NotEmpty final List<@NotBlank String> ids) {
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/SelectorController.java:145]: the api is error, the api is batch operation

Comment (moremind) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/SelectorController.java:145]: you should delete selected ids in namespace

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java [hunk 12]
```diff
@@ -65,20 +65,19 @@ public interface NamespacePluginRelMapper extends ExistProvider {
     /**
      * select plugin by namespacePluginId.
      *
-     * @param id    primary key.
+     * @param pluginId    primary key.
      * @param namespaceId namespace id.
      * @return {@linkplain PluginVO}
      */
-    NamespacePluginVO selectById(String id, String namespaceId);
+    NamespacePluginVO selectByPluginIdAndNamespaceId(String pluginId, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java:72]: this is id or pluginId?

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java [hunk 13]
```diff
@@ -44,13 +44,22 @@ public interface SelectorMapper extends ExistProvider {
...
+    SelectorDO selectByIdAndNamespaceId(String id, String namespaceId);
+
     /**
      * select selector by id.
      *
      * @param id primary key.
      * @return {@linkplain SelectorDO}
      */
-    SelectorDO selectById(String id);
+    SelectorDO selectByOnlyId(String id);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java:62]: just named selectById

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/model/entity/SelectorDO.java [hunk 14]
```diff
@@ -393,14 +421,15 @@ public static SelectorData transFrom(final SelectorDO selectorDO, final String p
...
     }
-    
+
     /**
      * selector data transform.
      *
-     * @param selectorDO selector entity
-     * @param pluginName plugin name
+     * @param selectorDO        selector entity
+     * @param pluginName        plugin name
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/model/entity/SelectorDO.java:432]: not change the code

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/model/event/plugin/BatchPluginDeletedEvent.java [hunk 15]
```diff
@@ -31,14 +31,14 @@
...
     private final List<String> deletedPluginIds;
-    
+
     /**
      * Create a new {@code PluginChangedEvent}.operator is unknown.
      *
-     * @param source   Current plugin state
-     * @param operator operator
+     * @param source      Current plugin state
+     * @param operator    operator
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/model/event/plugin/BatchPluginDeletedEvent.java:41]: not change the comment

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java [hunk 16]
```diff
@@ -107,55 +107,61 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
...
     /**
      * find selector by id.
      *
-     * @param id primary key.
+     * @param id          primary key.
+     * @param namespaceId namespaceId.
      * @return {@linkplain SelectorVO}
      */
-    SelectorVO findById(String id);
+    SelectorVO findById(String id, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:123]: named findByIdAndNamespaceId

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java [hunk 17]
```diff
@@ -107,55 +107,61 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
     /**
      * delete selectors.
      *
-     * @param ids primary key.
+     * @param ids         primary key.
+     * @param namespaceId namespaceId.
      * @return rows int
      */
-    int delete(List<String> ids);
+    int delete(List<String> ids, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:114]: change method name

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java [hunk 18]
```diff
@@ -107,55 +107,61 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
...
+     * @param name        the name
+     * @param namespaceId namespaceId.
+     *                    Therefore, it is recommended to: {@linkplain SelectorService#findListByName(java.lang.String, java.lang.String)}
      * @return selector do
      * @deprecated sice 2.6.0  Deprecated. By querying under this condition, multiple data are usually obtained.
-     *              Therefore, it is recommended to: {@linkplain SelectorService#findListByName(java.lang.String)}
      */
     @Deprecated
-    SelectorDO findByName(String name);
+    SelectorDO findByName(String name, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:136]: change name

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java [hunk 19]
```diff
@@ -107,55 +107,61 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
...
     /**
      * find selector list by name.
      *
-     * @param name name
+     * @param name        name
+     * @param namespaceId namespaceId.
      * @return list
      */
-    List<SelectorDO> findListByName(String name);
+    List<SelectorDO> findListByName(String name, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:144]: change name

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java (hunk 7 above)
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:154]: change name

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java [hunk 20]
```diff
@@ -107,55 +107,61 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
...
      *
-     * @param name       name
-     * @param pluginName pluginName
+     * @param name        name
+     * @param pluginName  pluginName
+     * @param namespaceId namespaceId.
      * @return SelectorDO
      */
-    SelectorDO findByNameAndPluginNameForUpdate(String name, String pluginName);
+    SelectorDO findByNameAndPluginNameForUpdate(String name, String pluginName, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:164]: change name

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java [hunk 21]
```diff
@@ -202,10 +210,11 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
     /**
      * Find by plugin id list.
      *
-     * @param pluginId the plugin id
+     * @param pluginId    the plugin id
+     * @param namespaceId the namespaceId
      * @return the list
      */
-    List<SelectorData> findByPluginId(String pluginId);
+    List<SelectorData> findByPluginId(String pluginId, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:217]: please modify the same question.

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/DiscoveryUpstreamServiceImpl.java [hunk 22]
```diff
@@ -154,7 +154,7 @@ public List<DiscoverySyncData> listAll() {
             if (StringUtils.hasLength(discoveryRelDO.getSelectorId())) {
                 String selectorId = discoveryRelDO.getSelectorId();
                 discoverySyncData.setSelectorId(selectorId);
-                SelectorDO selectorDO = selectorMapper.selectById(selectorId);
+                SelectorDO selectorDO = selectorMapper.selectByOnlyId(selectorId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/DiscoveryUpstreamServiceImpl.java:157]: not change

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/PluginServiceImpl.java [hunk 23]
```diff
@@ -130,6 +135,12 @@ public String delete(final List<String> ids) {
         if (CollectionUtils.isEmpty(plugins)) {
             return AdminConstants.SYS_PLUGIN_ID_NOT_EXIST;
         }
+        Optional<PluginDO> exist = plugins.stream()
+                .filter(value -> !Objects.isNull(this.namespacePluginRelMapper.selectByPluginId(value.getId())))
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/PluginServiceImpl.java:139]: use Objects.nonNull()

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/PluginServiceImpl.java [hunk 24]
```diff
@@ -130,6 +135,12 @@ public String delete(final List<String> ids) {
...
+                .findAny();
+        if (exist.isPresent()) {
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/PluginServiceImpl.java:141]: why findAny?

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/UpstreamCheckService.java [hunk 25]
```diff
@@ -437,7 +439,8 @@ public void fetchUpstreamData() {
         }
         Map<String, String> pluginMap = pluginDOList.stream().filter(Objects::nonNull)
                 .collect(Collectors.toMap(PluginDO::getId, PluginDO::getName, (value1, value2) -> value1));
-        final List<SelectorDO> selectorDOList = selectorMapper.findByPluginIds(new ArrayList<>(pluginMap.keySet()));
+        // todo:[To be refactored with namespace] Temporarily hardcode
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/UpstreamCheckService.java:442]: if complete, please remove todo

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/publish/NamespacePluginEventPublisher.java [hunk 26]
```diff
@@ -77,9 +80,21 @@ public void onDeleted(final NamespacePluginVO namespacePluginVO) {
...
     @Override
     public void onDeleted(final Collection<NamespacePluginVO> namespacePlugin) {
-        publish(new BatchNamespacePluginChangedEvent(namespacePlugin, null, EventTypeEnum.PLUGIN_UPDATE, SessionUtil.visitorName()));
-        publisher.publishEvent(new DataChangedEvent(ConfigGroupEnum.PLUGIN, DataEventTypeEnum.UPDATE,
+        String namespaceId = ((Collection<?>) namespacePlugin)
+                .stream()
+                .map(NamespacePluginVO.class::cast)
+                .findFirst()
+                .map(NamespacePluginVO::getNamespaceId)
+                .orElse("");
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/publish/NamespacePluginEventPublisher.java:88]: why orElse("")???

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java (hunk 13 above)
Comment (xcsnx) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java:62]: ok

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/UpstreamCheckService.java (hunk 25 above)
Comment (xcsnx) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/UpstreamCheckService.java:442]: not complete

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java (hunk 12 above)
Comment (xcsnx) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java:72]: pluginId

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/DiscoveryUpstreamMapper.java [hunk 27]
```diff
@@ -38,15 +38,15 @@ public interface DiscoveryUpstreamMapper extends ExistProvider {
     Boolean existed(@Param("id") Serializable key);
 
     /**
-     * selectByIds.
+     * selectByIdsAndNamespaceId.
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/DiscoveryUpstreamMapper.java:41]: why change comment?

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java [hunk 28]
```diff
@@ -65,20 +65,19 @@ public interface NamespacePluginRelMapper extends ExistProvider {
     /**
      * select plugin by namespacePluginId.
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java:66]: change the comment

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java [hunk 29]
```diff
@@ -124,22 +123,22 @@ public interface NamespacePluginRelMapper extends ExistProvider {
      * @param namespaceId namespace id.
      * @return rows int
      */
-    int deleteByIds(List<String> ids, String namespaceId);
+    int deleteByIdsAndNamespaceId(List<String> ids, String namespaceId);
 
     /**
      * select all.
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java:129]: change the comment

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java [hunk 30]
```diff
@@ -124,22 +123,22 @@ public interface NamespacePluginRelMapper extends ExistProvider {
...
      *
      * @param namespaceId namespace id.
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/NamespacePluginRelMapper.java:131]: same question

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java [hunk 31]
```diff
@@ -44,6 +44,15 @@ public interface SelectorMapper extends ExistProvider {
     @Override
     Boolean existed(@Param("id") Serializable id);
 
+    /**
+     * select selector by id.
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java:48]: change the comment

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java [hunk 32]
```diff
@@ -199,7 +214,7 @@ public interface SelectorMapper extends ExistProvider {
     /**
      * update status.
      *
-     * @param id id
+     * @param id      id
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java:217]: not modify the style

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java [hunk 33]
```diff
@@ -69,37 +78,43 @@ public interface SelectorMapper extends ExistProvider {
...
+     * @param name        the name
+     * @param namespaceId namespaceId.
      * @return selector do list
      */
-    List<SelectorDO> selectByName(String name);
+    List<SelectorDO> selectByNameAndNamespaceId(String name, String namespaceId);
 
     /**
-     * Find by name and plugin id selector do.
+     * Find selector do by name and plugin id and namespaceId.
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/SelectorMapper.java:108]: selector entity

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java [hunk 34]
```diff
@@ -105,57 +105,63 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
...
+     * @param pluginName  the plugin name
+     * @param namespaceId the namespaceId
      * @return the selector do
      */
-    SelectorDO findByNameAndPluginName(String name, String pluginName);
+    SelectorDO findByNameAndPluginNameAndNamespaceId(String name, String pluginName, String namespaceId);
 
     /**
-     * findByNameAndPluginNameForUpdate.
+     * findByNameAndPluginNameAndNamespaceIdForUpdate.
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java:157]: use normal comment. `find selector entity by selector name, plugin name and namespace id`

**Review by moremind — CHANGES_REQUESTED** [review]

**Review by moremind — APPROVED** [review]

**Review by moremind — APPROVED** [review]

**Pull request merged**